overlord/snapstate/handlers_components.go

```diff
@@ -312,17 +313,12 @@ func (m *SnapManager) undoLinkComponent(t *state.Task, _ *tomb.Tomb) error {
 
 	// Restore old state
 	// relinking of the old component is done in the undo of unlink-current-snap
-	restoredComps := []*snap.ComponentSideInfo{}
-	for _, csi := range revSt.Components {
-		if csi.Equal(linkedComp) {
-			continue
-		}
-		restoredComps = append(restoredComps, csi)
-	}
-	revSt.Components = restoredComps
+	snapSt.Sequence.RemoveComponentForRevision(snapInfo.Revision, linkedComp.Component)
 
 	// Finally, write the state
 	Set(st, snapsup.InstanceName(), snapSt)
+	// Make sure we won't be rerun
+	t.SetStatus(state.DoneStatus)
```

Comment:
this needs to be Undone as we are an undo handler

Reply:
Right, changed now